tests/storybook.ts

```diff
@@ -1,12 +1,22 @@
-import { runInRepo } from '../utils.ts'
+import { $, runInRepo } from '../utils.ts'
 import type { RunOptions } from '../types.d.ts'
+import { getCommand } from '@antfu/ni'
 
 export async function test(options: RunOptions) {
 	await runInRepo({
 		...options,
-		repo: 'storybookjs/builder-vite',
-		branch: 'main',
-		build: 'prepublish',
-		test: ['build-examples', 'test-ci'],
+		repo: 'storybookjs/sandboxes',
+		dir: 'next',
+		testSubdirectory: 'react-vite/default-ts/after-storybook',
+		branch: 'next',
+		build: ['build-storybook'],
+		beforeTest: async (agent) => {
+			$`${getCommand(agent, 'execute', ['http-server', 'storybook-static', '--port 6006', '--silent'])}`
+			await $`${getCommand(agent, 'execute', ['wait-on', 'http://localhost:6006'])}`
+		},
+		test: async (agent) => {
+			await $`${getCommand(agent, 'execute', ['@storybook/test-runner'])}`
+			await $`${getCommand(agent, 'execute', ['fkill-cli', ':6006'])}`
+		},
```

Comment:
Thanks for working on this PR! It would be great to have Storybook in our CI again!

About these scripts, having the `agent` looks ok to me to handle edge cases but the details in `beforeTest` and `test` will end up creating maintenance issues because you'll need to update them here instead of being in sync with your repo. Would it be possible to have a `test:vite` script in your repo that does all this so you can maintain it and change it independently of vite-ecosystem-ci?

Reply:
The https://github.com/storybookjs/sandboxes repository is used to build Storybook for different frameworks and renderer combinations. It is updated once per day. We run a script like `npm create vite` and initialize Storybook on top of it. Our scripts are currently not very flexible enough to hook in after build and, for example, adjust the package.json of a particular sandbox and set everything up to provide a `test:vite` script command.

Second, our users also use these sandboxes to provide reproductions. So, creating a script like `test:vite` would expose it to our users as well, which we potentially don't want to do.

I understand your concerns! I am optimistic that how a working Storybook can be tested will not change frequently. Running Storybook on the port `6006` wasn't ever changed and is unlikely to be changed. The test-runner is also a package that we maintain, and the CLI should remain stable as it is.

If you still have concerns, I suggest adjusting our sandbox generation scripts to hook in after the build. But again, the sandboxes should mirror real-world scenarios, and there, the `@storybook/test-runner`, for example, isn't set up per default.

Reply:
This sounds ok to me for now then. @dominikg, what do you think? If we see this is causing issues we can rethink how we are integrating with Storybook later

Reply:
i'd still prefer this to be encapsulated in a script controlled on the generated projects side. Long term we need to redesign ecosystem-ci to have less configuration about the projects under test in this repo.

Is there a way for the storybook testrunner to start/stop the webserver itself similar to playwright? https://playwright.dev/docs/test-webserver  ? That could reduce the calls here to a single one.

the hardcoded port 6006 (which i believe is the default port of storybook) could also be a problem when running ecosystem-ci locally. This is needed to be able to reproduce or bisect issues.